### PR TITLE
libopenmpt: default to supporting pulseaudio

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change

see: https://github.com/NixOS/nixpkgs/issues/138332

When #136735 was merged, it included this as a new input for libopenmpt:

`usePulseAudio ? config.pulseaudio or false`

Any user that had `config.pulseaudio = true` set, will start having to build libopenmpt and its dependents (including webkit-gtk) from source.

We should instead default to building this with pulse support to be more in line with other desktop-oriented packages in nixpkgs that tend to build with support for Pulse when building for linux.

For example, this is how `libao` is packaged+built.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by colemickens** [comment]
cc: @SuperSandro2000

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -26475,7 +26475,9 @@ with pkgs;
 
   vivaldi-widevine = callPackage ../applications/networking/browsers/vivaldi/widevine.nix { };
 
-  libopenmpt = callPackage ../applications/audio/libopenmpt { };
+  libopenmpt = callPackage ../applications/audio/libopenmpt {
+    usePulseAudio = config.pulseaudio or stdenv.isLinux;
```
Comment (OPNA2608) [inline pkgs/top-level/all-packages.nix:26479]: Why did you move this to all-packages.nix?

Comment (colemickens) [reply pkgs/top-level/all-packages.nix:26479]: hm, I guess `stdenv` is available in there.

1. that's what (some) other packages like `libao` 
2. I thought it was related to deps like this that might only be enabled on certain platforms, and thus that determination might be made at the all-packages level?

I'm not sure, I can change the patch to just be `s/false/stdenv.isLinux`.

Comment (OPNA2608) [reply pkgs/top-level/all-packages.nix:26479]: Most packages I've worked with do these kinds of platform-specific decisions in the package's code instead of the all-packages level. I'd prefer to stick with this & make it `s/false/stdenv.isLinux`.

Comment (colemickens) [reply pkgs/top-level/all-packages.nix]: Ack! Done and pushed!

**Comment by nrdxp** [comment]
Any idea how this would affect pipewire users, since they seem to be growing in number?

**Comment by colemickens** [comment]
Basically, this PR is only changing whether this particular lib has PulseAudio support built in natively. Without this PR, this particular lib is, by default, not built with native Pulse support. However, if it uses ALSA, things will still work (*) for Pulse users due to how ALSA/pulse are configured in NixOS.

If you use PipeWire, presumably you want native PipeWire support, and short of that, native Pulse support.

After this is merged, I'll probably open issues for looking at removing the config option entirely and defaulting to building with pulse everywhere. If per-packages want to make it configurable, thats one thing, but a global package overide that (somewhat needlessly) splits nixpkgs in two... Seems bad.

(*): This is what I was told, and must somewhat be true given how many things seem to be built in nixpkgs without pulse by default.
